### DISCUSSION TITLE
Bulk trim transparent pixels !!! only on two sides of the image !!!

### DISCUSSION THREAD

**Discussion by kolyafromburgerking** [ImageMagick/ImageMagick#6982] [Digital Image Processing] [answered]
I have hundreds of images like this one, and what I need to do is, while not touching the sides, trim transparent pixels above and below the image.
![image](https://github.com/ImageMagick/ImageMagick/assets/137233302/2b4ab4b3-8ab1-46a5-b5d4-614deb55733c)
Basically, I don't just want the width of the image to stay the same, I want the opaque part to retain its horizontal position. I've searched for similar things like this but have only found how to do this on all sides, which wouldn't work for my case.

I'm not confident in my explaining abilities, so here's an illustrated example.
![image](https://github.com/ImageMagick/ImageMagick/assets/137233302/27e22986-b694-43aa-a568-e01b1e81bc1b)
P.S. i have zero imagemagick scripting experience

**Comment by fmw42** [comment, marked as answer]
The first image you posted has only one flashlight. There is no way to trim any parts of it without trimming its sides and making it smaller.  Please post an example where there is a region that can be trimmed as you describe.

I do not see how one can trim the area you marked in your second image with a green check mark without making the image smaller vertically nor keeping is vertical positioning.

Are you ok with vertical modifications, but just want the horizontals to be preserved?  So in the top image, you want to trim the top and bottom transparent regions, but not the sides? Is that correct?

The solution for that case image, only would be to splice a one pixel color column to one side and trim, then do the same to the other side and trim.  See https://imagemagick.org/Usage/crop/#trim_oneside


Input:

![img](https://github.com/ImageMagick/ImageMagick/assets/13458218/fe3e0f60-02f6-49a4-af8c-8ce525989cbc)

`convert img.png -background red -gravity north -splice 0x1 -trim +repage -gravity south -splice 0x1 -trim +repage img2.png`

![img2](https://github.com/ImageMagick/ImageMagick/assets/13458218/26fdcb4e-ac0e-4f9c-9aed-c445537824fc)

The input has dimension of 302 × 1054 and the output has dimensions of 302 × 44.

The above works for Imagemagick 7 by changing convert to magick.

On Imagemagick 7, you can also do

`magick img.png -define trim:edges=north,south -trim +repage img2.png`

and get the same result.


This does not work for internal transparent regions between multiple opaque regions.

**Reply by kolyafromburgerking** [reply]
The `trim:edges=north,south` command was the exact thing i was looking for, thank you!!

This is what i used for batch conversion
```
mkdir edited\images\dir
cd raw\images\dir
magick mogrify -define trim:edges=north,south -trim +repage -path edited\images\dir *.png
```
